test/disjointPoolFileProv.cpp

```diff
@@ -0,0 +1,363 @@
+    umf::providerMakeCOps<umf_test::provider_ba_global, void>();
+
+struct FileWithMemoryStrategyTest
+    : umf_test::test,
+      ::testing::WithParamInterface<coarse_strategy_t> {
+    void SetUp() override {
+        test::SetUp();
+        allocation_strategy = this->GetParam();
+    }
+
+    coarse_strategy_t allocation_strategy;
+};
+
+INSTANTIATE_TEST_SUITE_P(
+    FileWithMemoryStrategyTest, FileWithMemoryStrategyTest,
+    ::testing::Values(UMF_COARSE_MEMORY_STRATEGY_FASTEST,
+                      UMF_COARSE_MEMORY_STRATEGY_FASTEST_BUT_ONE,
+                      UMF_COARSE_MEMORY_STRATEGY_CHECK_ALL_SIZE));
+
+TEST_P(FileWithMemoryStrategyTest, disjointFileMallocPool_simple1) {
+    umf_memory_provider_handle_t malloc_memory_provider;
+    umf_result_t umf_result;
+
+    umf_result = umfMemoryProviderCreate(&UMF_MALLOC_MEMORY_PROVIDER_OPS, NULL,
+                                         &malloc_memory_provider);
+    ASSERT_EQ(umf_result, UMF_RESULT_SUCCESS);
+    ASSERT_NE(malloc_memory_provider, nullptr);
+
+    umf_file_memory_provider_params_handle_t file_params = nullptr;
+    umf_result = umfFileMemoryProviderParamsCreate(&file_params, FILE_PATH);
+    ASSERT_EQ(umf_result, UMF_RESULT_SUCCESS);
+    ASSERT_NE(file_params, nullptr);
+
+    umf_memory_provider_handle_t file_memory_provider;
+    umf_result = umfMemoryProviderCreate(umfFileMemoryProviderOps(),
+                                         file_params, &file_memory_provider);
+    ASSERT_EQ(umf_result, UMF_RESULT_SUCCESS);
+    ASSERT_NE(file_memory_provider, nullptr);
+
+    umf_result = umfFileMemoryProviderParamsDestroy(file_params);
+    ASSERT_EQ(umf_result, UMF_RESULT_SUCCESS);
+
+    umf_disjoint_pool_params_handle_t disjoint_pool_params = NULL;
+    umf_result = umfDisjointPoolParamsCreate(&disjoint_pool_params);
+    ASSERT_EQ(umf_result, UMF_RESULT_SUCCESS);
+    ASSERT_NE(disjoint_pool_params, nullptr);
+    umf_result =
+        umfDisjointPoolParamsSetSlabMinSize(disjoint_pool_params, 4096);
+    ASSERT_EQ(umf_result, UMF_RESULT_SUCCESS);
+    umf_result =
+        umfDisjointPoolParamsSetMaxPoolableSize(disjoint_pool_params, 4096);
+    ASSERT_EQ(umf_result, UMF_RESULT_SUCCESS);
+    umf_result = umfDisjointPoolParamsSetCapacity(disjoint_pool_params, 4);
+    ASSERT_EQ(umf_result, UMF_RESULT_SUCCESS);
+    umf_result =
+        umfDisjointPoolParamsSetMinBucketSize(disjoint_pool_params, 64);
+    ASSERT_EQ(umf_result, UMF_RESULT_SUCCESS);
+    umf_result = umfDisjointPoolParamsSetTrace(disjoint_pool_params, 1);
+    ASSERT_EQ(umf_result, UMF_RESULT_SUCCESS);
+
+    umf_memory_pool_handle_t pool;
+    umf_result =
+        umfPoolCreate(umfDisjointPoolOps(), file_memory_provider,
+                      disjoint_pool_params, UMF_POOL_CREATE_FLAG_NONE, &pool);
+    ASSERT_EQ(umf_result, UMF_RESULT_SUCCESS);
+    ASSERT_NE(pool, nullptr);
+
+    umf_result = umfDisjointPoolParamsDestroy(disjoint_pool_params);
+
+    umf_memory_provider_handle_t prov = NULL;
```

Comment:
In C++ files, `nullptr` is preferred for any pointer type.

Reply:
Done